src/wwwroot/controllers/PlanCtrl.js

```diff
@@ -174,6 +184,42 @@
       }      
     }
 
+    function isUpgradeablePlan(limit){
+      //Check constraints to validate that plan is upgradeable
+
+      //Set to true all errors to prevent access if service is not working
+      var errorsPlanUpgrade = {'isDKIMError': true, "isDeliveredError": true, "isDeliveredAndDKIMError": true};
+
+      if(limit==undefined){
+        var isConfigRequired = auth.getLimitsByAccount().then(isUpgradeablePlan).catch($log.error);
+      }
+      else{
+        //Exist activity in the current plan?
+        errorsPlanUpgrade.isDeliveredError = vm.totalDeliveriesCount > 0? false : true;
+        //It is the domain correctly configurated?
+        errorsPlanUpgrade.isDKIMError = limit.requiresDomainConfiguration; 
+        //This account is not registering deliveries and is not correctly configurated?
+        errorsPlanUpgrade.isDeliveredAndDKIMError = errorsPlanUpgrade.isDeliveredError && errorsPlanUpgrade.isDKIMError;
+      }
+
+      return errorsPlanUpgrade;
+    }
+
+    function isValidUpgradeablePlan(){
+      var haveErrors = (vm.errorsPlanUpgrade.isDKIMError || vm.errorsPlanUpgrade.isDeliveredError || vm.errorsPlanUpgrade.isDeliveredAndDKIMError)? true : false;
+      return !haveErrors;
+    }
+
+    $scope.disableNotUpgradable=function(event){
```

Comment:
Check that there are controllers with Sintax "Controller as vm", there are several benefits from using "scope".

Reply:
Thanks @mvillaseco I will apply this change in the next commit.